mitmproxy/addons/proxyserver.py

```diff
@@ -259,14 +259,21 @@ def configure(self, updated) -> None:
                     )
 
             # ...and don't listen on the same address.
-            listen_addrs = [
-                (
-                    m.listen_host(ctx.options.listen_host),
-                    m.listen_port(ctx.options.listen_port),
-                    m.transport_protocol,
-                )
-                for m in modes
-            ]
+            listen_addrs = []
+            for m in modes:
+                protocols = []
+                if m.transport_protocol == "both":
+                    protocols.extend(["tcp", "udp"])
+                else:
+                    protocols.append(m.transport_protocol)
```

Comment:
```suggestion
                if m.transport_protocol == "both":
                    protocols = ["tcp", "udp"]
                else:
                    protocols = [m.transport_protocol]
```

Reply:
We get a mypy error with this if-else pattern
`error: List item 0 has incompatible type "Literal['tcp', 'udp'] | None"; expected "str"  [list-item]`
so I've reversed the conditions
We could instead set the variable type using `protocols: list[Literal['tcp', 'udp'] | None]`, lmk if that's better

Reply:
Ah, thanks. The mypy complaint isn't completely unjustified, `m.transport_protocol` can be None by it's type annotation. It's only set to None for local mode, and that comes from a time where "both" was not a valid option yet.  Let's remove the `| None` part for transport_protocol, make it `both` for Local/OsProxy mode, and this error should go away? 
(If it doesn't, a `protocols: list[TransportProtocol]` line before the if/else should fix it.)